finetune/finetune_power.py

```diff
@@ -36,27 +39,10 @@ def setup_model(model_type: str, device: torch.device) -> torch.nn.Module:
     """
     if model_type == "PanguPowerPatchRecovery":
         model = PanguPowerPatchRecovery(device=device).to(device)
-
-        checkpoint = torch.load(
-            cfg.PG.BENCHMARK.PRETRAIN_24_torch, map_location=device, weights_only=False
-        )
-
-        pretrained_dict = checkpoint["model"]
-        model_dict = model.state_dict()
-
-        # Filter out keys in pretrained_dict that belong to _output_layer (conv and conv_surface)
-        pretrained_dict = {
-            k: v for k, v in pretrained_dict.items() if "_output_layer" not in k
-        }
-
-        # Update the model's state_dict except the _output_layer
-        model_dict.update(pretrained_dict)
-
-        # Load the updated state_dict into the model
-        model.load_state_dict(model_dict)
+        model.load_pangu_state_dict(device)
 
         # Only finetune the last layer
-        set_requires_grad(model, "_output_layer")
+        set_requires_grad(model, "_output_power_layer")
 
     elif model_type == "PanguPowerConv":
```

Comment:
Maybe include a brief comment describing what this model type means, i.e. what are it's characteristics etc? Or is this something you will detail more in your main repository?

Reply:
I now added descriptions of each of the models in pangu_power :)